### ISSUE TITLE
Highlight time slots of same group

### ISSUE THREAD

**Issue by madsonic** [nusmodifications/nusmods#754]
**Do you want to request a *feature* or report a *bug*?**

UI Feature (enhancement)

**What is the current behavior?**

Selection of time slot displays all possibilities without any group highlights 

**What is the expected behavior?**

Highlight sessions of the same group during time slot selection. By group I mean different sessions of the same tutorial/lab in the week (see GEK1549 E01 in the edited screen shot). Otherwise it is especially difficult to find the matching sessions if there are numerous possibilities as with a module like this. More than happy to help with this enhancement.

While hovering over other choices the same groups should have a different color. The same effect might not be easy to product on mobile since there is no 'hover'

![group highlight demo](https://user-images.githubusercontent.com/437672/35137683-b058b63e-fd25-11e7-884c-63e19a5370fc.png)

**Further information**

- Which version of NUSMods are you currently using? (Check the footer for the version number, 201XXXXX-XXXXXXX)
20180117-9ba62bd
- Did this work in NUSMods BEFORE the R release/redesign? Yes/No/Unsure
No
- Did this work in NUSMods AFTER the R release/redesign? Yes/No/Unsure
No
- Which browser and version are you using? E.g. Chrome 62/Firefox Quantum 57/Safari 11.1.1/etc
Version 63.0.3239.132 (Official Build) (64-bit)
- Which device are you using? E.g. MacBook Pro/Surface Pro/iPhone X/etc
MacBook Pro
- Which operating system and version are you on? E.g. macOS 10.12.5/Windows 10/iOS 11.0.0/etc
macOS 10.12.6 (16G1114)

**Comment by madsonic** [comment]
Cool. Do you mind assigning this to me? I shall play around with it for a bit first. Will let you if I have questions

**Comment by ZhangYiJiang** [comment]
Absolutely. You can join us in our Telegram group if you want to talk to us.

**Comment by Rinder5** [comment]
@madsonic how's the progress? I also have a few ideas I want to try out =P

**Comment by madsonic** [comment]
@Rinder5 I was occupied by school work for awhile and didnt manage to do anything meaningful for this issue. perhaps you could give it a shot!

**Comment by Rinder5** [comment]
alright, thank you! =)
